Runtime/Properties/HighlightProperty.cs

```diff
@@ -29,6 +29,28 @@ protected override void OnEnable()
                 Highlighter = GetComponent<InteractableHighlighter>();
             }
         }
+        
+        protected void Reset()
+        {
+            InteractableObject ownInteractableObject = gameObject.GetComponent<InteractableObject>();
+            Rigidbody ownRigidbody = gameObject.GetComponent<Rigidbody>();
+            
+            // If gameObject was not interactable before, disable interactable functionality.
+            if (ownInteractableObject == null)
+            {
+                ownInteractableObject = gameObject.GetOrAddComponent<InteractableObject>();
+                ownInteractableObject.IsGrabbable = false;
+                ownInteractableObject.IsTouchable = false;
+                ownInteractableObject.IsUsable = false;
+                // If it had no rigidbody and thus was unaffected by physics, make it kinematic.
+                if (ownRigidbody == null)
+                {
+                    gameObject.GetOrAddComponent<Rigidbody>().isKinematic = true;
```

Comment:
There is no need to get the rigidbody before this line, neither to see it it exists.

Reply:
We need this check because we do not want to make it kinematic if it already exists. (Example: You already have the GrabbableProperty and add the HighlightProperty afterwards -> the object should still be interactable and thus non-kinematic)

Reply:
And line 41 adds the `InteractableObject` component which by default adds a RigidBody if there is none. So we need to know beforehand, if there is already an Rigidbody which must not be made changed.

Reply:
Ok, at least getting the Rigidbody can be done inside the if.

Reply:
Done.